### PR TITLE
Added wrap_dataset_for_transforms_v2 into datasets and handled beta w…

### PR DESCRIPTION
**Pull request by vfdev-5**
Description:
- Added wrap_dataset_for_transforms_v2 into datasets and handled beta warning 
- Added tests

Usage:
```
python -c "import torchvision; torchvision.disable_beta_transforms_warning(); from torchvision.datasets import wrap_dataset_for_transforms_v2"
> 
```

```
python -c "from torchvision.datasets import wrap_dataset_for_transforms_v2"
> 
/vision/torchvision/transforms/v2/__init__.py:54: UserWarning: The torchvision.datapoints and torchvision.transforms.v2 namespaces are still Beta. While we will try our best to maintain backward compatibility, some APIs or behaviors might change without a deprecation cycle. To help us improve these new features, please provide your feedback here: https://github.com/pytorch/vision/issues/6753.You can silence this warning by calling torchvision.disable_beta_transform_warning().
  warnings.warn(_BETA_TRANSFORMS_WARNING)
/vision/torchvision/datapoints/__init__.py:14: UserWarning: The torchvision.datapoints and torchvision.transforms.v2 namespaces are still Beta. While we will try our best to maintain backward compatibility, some APIs or behaviors might change without a deprecation cycle. To help us improve these new features, please provide your feedback here: https://github.com/pytorch/vision/issues/6753.You can silence this warning by calling torchvision.disable_beta_transform_warning().
  warnings.warn(_BETA_TRANSFORMS_WARNING)
```

```
python -c "from torchvision.datasets import CIFAR10"
> 
```

### CODE REVIEW

File: test/test_v2_dataset_warnings.py [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+
+    setattr(torchvision, "_WARN_ABOUT_BETA_TRANSFORMS", True)
+
+    with pytest.warns(UserWarning, match=torchvision._BETA_TRANSFORMS_WARNING):
+        from torchvision.datasets import wrap_dataset_for_transforms_v2
+
+    setattr(torchvision, "_WARN_ABOUT_BETA_TRANSFORMS", value)
+
+
+def test_no_warns_if_imported_from_datasets():
```
Comment (NicolasHug) [inline test/test_v2_dataset_warnings.py:17]: This test is failing, but I'd suggest removing both tests.
They can interfer with the import and warnings in ways that we may not predict properly, and this may lead to surprising or undesired behaviour (like masking a warning it shouldn't mask)

File: test/test_v2_dataset_warnings.py [hunk 2]
```diff
@@ -0,0 +1,22 @@
+import pytest
+
+
+def test_warns_if_imported_from_datasets():
+    import torchvision
+
+    value = torchvision._WARN_ABOUT_BETA_TRANSFORMS
+
+    setattr(torchvision, "_WARN_ABOUT_BETA_TRANSFORMS", True)
```
Comment (pmeier) [inline test/test_v2_dataset_warnings.py:9]: You can do that with `mocker.patch_object`: https://docs.python.org/3/library/unittest.mock.html#unittest.mock.patch.object

This way you don't have to reset in manually.

File: test/test_v2_dataset_warnings.py [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+
+    with pytest.warns(UserWarning, match=torchvision._BETA_TRANSFORMS_WARNING):
+        from torchvision.datasets import wrap_dataset_for_transforms_v2
+
+    setattr(torchvision, "_WARN_ABOUT_BETA_TRANSFORMS", value)
```
Comment (pmeier) [inline test/test_v2_dataset_warnings.py:14]: If you want to do it manually, you will need the context manager into a `try / finally` and reset there. Otherwise, you will pollute global state if the context manager errors out, i.e. no warning is raised.

File: test/test_v2_dataset_warnings.py [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+
+
+def test_no_warns_if_imported_from_datasets():
+    import warnings
+
+    with warnings.catch_warnings():
+        warnings.simplefilter("error")
+        from torchvision.datasets import wrap_dataset_for_transforms_v2
```
Comment (pmeier) [inline test/test_v2_dataset_warnings.py:22]:
```suggestion
@pytest.mark.filterwarnings("error")
def test_no_warns_if_imported_from_datasets():
    from torchvision.datasets import wrap_dataset_for_transforms_v2
```

**Review by pmeier — COMMENTED** [review]
Just a few comments on the tests. Module level `__getattr__` for the import looks fine to me to resolve the circular dependency.

**Comment by pmeier** [comment]
Test are failing though: https://github.com/pytorch/vision/actions/runs/4202967731/jobs/7291773255#step:10:33509

Same for the lint job: https://app.circleci.com/pipelines/github/pytorch/vision/23383/workflows/5bcb63e6-c258-47a8-8f66-cd09db2a43bb/jobs/1834284

**Review by NicolasHug — APPROVED** [review]
Thanks a lot for the PR @vfdev-5 . Some minor comment but the behaviour LGTM. We should also remove `wrap_dataset_for_transforms_v2` from `datapoints.__init__.py` before merging

File: torchvision/datasets/__init__.py [hunk 5]
```diff
@@ -128,3 +128,12 @@
     "InStereo2k",
     "ETH3DStereo",
 )
+
+
+def __getattr__(name):
```
Comment (NicolasHug) [inline torchvision/datasets/__init__.py:133]: I'd suggest linking to the SLEP and explain why we need this (i.e. to avoid raising a warning when importing other stuff from `torchvision.datasets`

**Review by pmeier — APPROVED** [review]
LGTM is CI is green. Thanks Victor!

File: .github/workflows/test-linux-cpu.yml [hunk 6]
```diff
@@ -55,3 +55,6 @@ jobs:
         # Run Tests
         python3 -m torch.utils.collect_env
         python3 -m pytest --junitxml=test-results/junit.xml -v --durations 20
+
+        # Specific test for warnings on "from torchvision.datasets import wrap_dataset_for_transforms_v2"
```
Comment (NicolasHug) [inline .github/workflows/test-linux-cpu.yml:59]:
```suggestion
        # Specific test for warnings on "from torchvision.datasets import wrap_dataset_for_transforms_v2"
        # We keep them separate to avoid any side effects due to warnings / imports.
        # TODO: Remove this and add proper tests (possibly using a sub-process solution as described
        # in https://github.com/pytorch/vision/pull/7269).
```

**Review by NicolasHug — APPROVED** [review]
Thanks a lot @vfdev-5 , still LGTM!
